SC2028 false negative for

#### For bugs
- Rule Id (if any, e.g. SC1000): SC2028
- My shellcheck version (`shellcheck --version` or "online"): `0.7.2`
- [x] The rule's wiki page does not already cover this (e.g. https://shellcheck.net/wiki/SC2086)
- [x] I tried on https://www.shellcheck.net/ and verified that this is still a problem on the latest commit

#### For new checks and feature suggestions
- [x] https://www.shellcheck.net/ (i.e. the latest commit) currently gives no useful warnings about this
- [x] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

```sh
#!/usr/bin/sh
echo "Case:\tTrue positive"
echo "Case:\e[41mFalse negative"
```

#### Here's what shellcheck currently says:
```
Line 3:
echo "Case:\tTrue positive"
     ^-- SC2028 (info): echo may not expand escape sequences. Use printf.
```

#### Here's what I wanted or expected to see:
```
Line 3:
echo "Case:\tTrue positive"
     ^-- SC2028 (info): echo may not expand escape sequences. Use printf.

Line 5:
echo "Case:\e[41mFalse negative"
     ^-- SC2028 (info): echo may not expand escape sequences. Use printf.
```

#### Relevant code

The code only checks for `-r`, `-n`, `-t`.
However there are many more escapes like `\e`, `\0xxx`, `\a` etc.
I think shellcheck should warn on any of those escape unless `echo -e` is used as there is a high probability that they are erroneous.

https://github.com/koalaman/shellcheck/blob/88cdb4e2c9b45becb21bd02cd7b205d5bef8cb56/src/ShellCheck/Checks/Commands.hs#L477

--- a/src/ShellCheck/Checks/Commands.hs
+++ b/src/ShellCheck/Checks/Commands.hs
@@ -481,9 +481,16 @@ prop_checkUnusedEchoEscapes2 = verifyNot checkUnusedEchoEscapes "echo -e 'foi\\n
 prop_checkUnusedEchoEscapes3 = verify checkUnusedEchoEscapes "echo \"n:\\t42\""
 prop_checkUnusedEchoEscapes4 = verifyNot checkUnusedEchoEscapes "echo lol"
 prop_checkUnusedEchoEscapes5 = verifyNot checkUnusedEchoEscapes "echo -n -e '\n'"
+prop_checkUnusedEchoEscapes6 = verify checkUnusedEchoEscapes "echo '\\506'"
+prop_checkUnusedEchoEscapes7 = verify checkUnusedEchoEscapes "echo '\\5a'"
+prop_checkUnusedEchoEscapes8 = verifyNot checkUnusedEchoEscapes "echo '\\8a'"
+prop_checkUnusedEchoEscapes9 = verifyNot checkUnusedEchoEscapes "echo '\\d5a'"
+prop_checkUnusedEchoEscapes10 = verify checkUnusedEchoEscapes "echo '\\x4a'"
+prop_checkUnusedEchoEscapes11 = verify checkUnusedEchoEscapes "echo '\\xat'"
+prop_checkUnusedEchoEscapes12 = verifyNot checkUnusedEchoEscapes "echo '\\xth'"
 checkUnusedEchoEscapes = CommandCheck (Basename "echo") f
   where
-    hasEscapes = mkRegex "\\\\[rnt]"
+    hasEscapes = mkRegex "\\\\([rntabefv\\']|[0-7]{1,3}|x([0-9]|[A-F]|[a-f]){1,2})"
     f cmd =
         whenShell [Sh, Bash, Ksh] $
             unless (cmd `hasFlag` "e") $

